java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala

```diff
@@ -54,42 +81,137 @@ object LoadDataPlan {
-    val offlineAddress = "/tmp/test_dir"
+  def protoTypeToScalaType(dataType: DataType): types.DataType = {
+    dataType match {
+      case DataType.kBool => BooleanType
+      case DataType.kSmallInt => ShortType
+      case DataType.kBigInt => LongType
+      case DataType.kInt => IntegerType
+      case DataType.kFloat => FloatType
+      case DataType.kDouble => DoubleType
+      case DataType.kDate => DateType
+      case DataType.kTimestamp => LongType // in openmldb, timestamp format is int64
+      case DataType.kVarchar | DataType.kString => StringType
+      case e: Any => throw new UnsupportedHybridSeException(s"unsupported proto DataType $e")
+    }
+  }
 
+  def gen(ctx: PlanContext, node: PhysicalLoadDataNode): SparkInstance = {
     val inputFile = node.File()
+    var db = ctx.getConf.defaultDb
+    if (node.Db().nonEmpty) {
+      db = node.Db()
+    }
+    val table = node.Table()
     val spark = ctx.getSparkSession
 
-    // read input file
-    val (format, mode, options) = parseOptions(node)
-    logger.info("format {}, mode {}, options {}", format, mode, options)
-    val df = spark.read.options(options).format(format).load(inputFile)
+    // get storage online/offline
+    // NOTICE: do not add "openmldb.execute.mode" in OpenmldbBatchConfig
+    val storage = spark.conf.get("openmldb.execute.mode", "offline")
+    require(storage == "offline" || storage == "online")
+
+    // read settings
+    val (format, options, mode, slk) = parseOptions(node)
+    logger.info("load data to storage {}, read[format {}, options {}], write[mode {}], is soft? {}", storage, format,
+      options, mode, slk.toString)
+
+    require(ctx.getOpenmldbSession != null, "LOAD DATA must use OpenmldbSession, not SparkSession")
+    val info = ctx.getOpenmldbSession.openmldbCatalogService.getTableInfo(db, table)
+    require(info != null && info.getName.nonEmpty, s"table $db.$table info is not existed(no table name): $info")
+    logger.info("table info: {}", info)
 
-    // write, offline address may contains some files, should choose the mode
-    df.write.mode(mode).parquet(offlineAddress)
+    // write
+    if (storage == "online") {
+      require(!slk && mode == "append", "import to online storage, can't do symbolic link, and mode must be append")
+
+      val writeOptions = Map("db" -> db, "table" -> table,
+        "zkCluster" -> ctx.getConf.openmldbZkCluster,
+        "zkPath" -> ctx.getConf.openmldbZkRootPath)
+      // The dataframe which be read should have the correct column types.
+      var struct = new StructType
+      DataType.getDescriptor
+      info.getColumnDescList.forEach(
+        col => struct = struct.add(col.getName, protoTypeToScalaType(col.getDataType), !col.getNotNull)
+      )
+      logger.info("read schema: {}", struct)
+      val df = spark.read.options(options).format(format).schema(struct).load(inputFile)
+      if (ctx.getConf.print) {
```

Comment:
We may remove this config later which is confusing and inefficient.

Reply:
ok, I'll use debug info to replace it